src/thing/WiFiManager.cpp

```diff
@@ -0,0 +1,69 @@
+#include "thing/WiFiManager.h"
+
+#include <ESP8266WiFi.h>
+#include <DNSServer.h>
+
+
+namespace thing {
+namespace {
+
+const short kDnsPort = 53;
+
+}  // namespace
+
+WiFiManager::WiFiManager() : debug_([](const char*){}) {}
+
+bool WiFiManager::StartAP() {
+  // Use arduino string, casting an int to std::string with arduino C++ support
+  // isn't easy.
```

Comment:
nope, not defined for some reason. There are other std:: things that arduino c++ doesn't support as well.